### PR TITLE
Laboratory work #3, Maxim Glukhankov - 22FPL1

### PR DESCRIPTION
**Pull request by glukhankov**

### CODE REVIEW

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_3_generate_by_ngrams/main.py [hunk 1]
```diff
@@ -45,24 +45,25 @@ def _tokenize(self, text: str) -> Optional[tuple[str, ...]]:
...
-            word_tokens = [alpha for alpha in word if alpha.isalpha()]
-            tokens.extend(word_tokens)
-            if word_tokens:
-                tokens.append(self._end_of_word_token)
+        tokenized_text = []
+        preprocessed_text = text.lower().split()
 
-        if not tokens:
+        for word in preprocessed_text:
+            tokens = [i for i in word if i.isalpha()]
```
[the remark this answers is not in the document]
Comment (glukhankov) [reply lab_3_generate_by_ngrams/main.py]: done

File: lab_3_generate_by_ngrams/main.py [hunk 2]
```diff
@@ -251,7 +249,7 @@ def _postprocess_decoded_text(self, decoded_corpus: tuple[str, ...]) -> Optional
 
         In case of corrupt input arguments, None is returned
         """
-        if not isinstance(decoded_corpus, tuple) or len(decoded_corpus) == 0:
+        if not isinstance(decoded_corpus, tuple) or not decoded_corpus:
```
[the remark this answers is not in the document]
Comment (glukhankov) [reply lab_3_generate_by_ngrams/main.py:243]: I tried to change them as much as I could

File: lab_3_generate_by_ngrams/main.py [hunk 3]
```diff
@@ -224,12 +222,12 @@ def _decode(self, corpus: tuple[int, ...]) -> Optional[tuple[str, ...]]:
...
         In case any of methods used return None, None is returned.
         """
-        if not isinstance(corpus, tuple) or len(corpus) == 0:
+        if not isinstance(corpus, tuple) or not corpus:
             return None
 
         decoded_corpus = []
-        for ident in corpus:
-            token = self.get_token(ident)
+        for i in corpus:
```
[the remark this answers is not in the document]
Comment (glukhankov) [reply lab_3_generate_by_ngrams/main.py:220]: thank you

[the remark this answers is not in the document]
Comment (artyomtugaryov) [reply lab_3_generate_by_ngrams/main.py:220]: but name is absolutelly bad

File: lab_3_generate_by_ngrams/main.py [hunk 4]
```diff
@@ -330,11 +320,13 @@ def build(self) -> int:
 
         for n_gram in n_grams:
             context_freq_dict[n_gram] = context_freq_dict.get(n_gram, 0) + 1
+        print(context_freq_dict)
```
Comment (artyomtugaryov) [inline lab_3_generate_by_ngrams/main.py:323]: remove

File: lab_3_generate_by_ngrams/main.py [hunk 5]
```diff
@@ -330,11 +320,13 @@ def build(self) -> int:
...
 
         lower_ngram_counts = {}
         for ngram, freq in context_freq_dict.items():
             context = ngram[:-1]
             lower_ngram_counts[context] = lower_ngram_counts.get(context, 0) + freq
+        print(lower_ngram_counts)
```
Comment (artyomtugaryov) [inline lab_3_generate_by_ngrams/main.py:329]: remove

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
What about start py? It contains code for 10

File: lab_3_generate_by_ngrams/main.py (hunk 4 above)
Comment (glukhankov) [reply lab_3_generate_by_ngrams/main.py]: done

File: lab_3_generate_by_ngrams/main.py (hunk 5 above)
Comment (glukhankov) [reply lab_3_generate_by_ngrams/main.py]: done

File: lab_3_generate_by_ngrams/main.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (artyomtugaryov) [reply lab_3_generate_by_ngrams/main.py:243]: If you cannot implement it - remove

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_3_generate_by_ngrams/main.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (glukhankov) [reply lab_3_generate_by_ngrams/main.py:243]: got it

**Comment by vitazaytseva** [comment]
Well done! Good job!

**Pull request closed without merging**